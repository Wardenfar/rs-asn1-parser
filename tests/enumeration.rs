use asn1_parser::parse_asn1_file;

#[test]
fn enum_ok() {
    let m = parse_asn1_file(include_str!("../resources/enum-ok.asn1")).unwrap();
    insta::assert_json_snapshot!(m);
}

#[test]
fn enum_err_1() {
    let e = parse_asn1_file(include_str!("../resources/enum-err-1.asn1")).unwrap_err();
    insta::assert_display_snapshot!(e)
}
