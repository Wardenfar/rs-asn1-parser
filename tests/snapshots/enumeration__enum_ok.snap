---
source: tests/enumeration.rs
assertion_line: 6
expression: m

---
{
  "name": {
    "name": "ModuleTestEnum1"
  },
  "header": {
    "keys": [
      {
        "NamedNumber": [
          {
            "name": "test"
          },
          5
        ]
      },
      {
        "Named": {
          "name": "abc"
        }
      }
    ]
  },
  "body": {
    "top_fields": [
      {
        "name": {
          "name": "Enum1"
        },
        "kind": {
          "Enum": {
            "items": [
              {
                "Name": {
                  "name": "red"
                }
              },
              {
                "Name": {
                  "name": "green"
                }
              },
              {
                "NamedNumber": [
                  {
                    "name": "blue"
                  },
                  4
                ]
              },
              {
                "Name": {
                  "name": "alpha"
                }
              }
            ]
          }
        }
      },
      {
        "name": {
          "name": "Enum2"
        },
        "kind": {
          "Enum": {
            "items": [
              {
                "Name": {
                  "name": "red"
                }
              },
              {
                "Name": {
                  "name": "green"
                }
              },
              {
                "NamedNumber": [
                  {
                    "name": "blue"
                  },
                  45
                ]
              },
              {
                "NamedNumber": [
                  {
                    "name": "orange"
                  },
                  23
                ]
              },
              {
                "Name": {
                  "name": "alpha"
                }
              }
            ]
          }
        }
      },
      {
        "name": {
          "name": "Enum3"
        },
        "kind": {
          "Enum": {
            "items": [
              {
                "Name": {
                  "name": "a"
                }
              },
              {
                "NamedNumber": [
                  {
                    "name": "b"
                  },
                  3
                ]
              },
              {
                "NamedNumber": [
                  {
                    "name": "c"
                  },
                  1
                ]
              }
            ]
          }
        }
      },
      {
        "name": {
          "name": "Enum4"
        },
        "kind": {
          "Enum": {
            "items": [
              {
                "Name": {
                  "name": "a"
                }
              },
              {
                "Name": {
                  "name": "b"
                }
              },
              {
                "NamedNumber": [
                  {
                    "name": "c"
                  },
                  3
                ]
              },
              {
                "NamedNumber": [
                  {
                    "name": "d"
                  },
                  4
                ]
              }
            ]
          }
        }
      },
      {
        "name": {
          "name": "Enum5"
        },
        "kind": {
          "Enum": {
            "items": [
              {
                "Name": {
                  "name": "a"
                }
              },
              {
                "NamedNumber": [
                  {
                    "name": "z"
                  },
                  25
                ]
              },
              {
                "Name": {
                  "name": "d"
                }
              }
            ]
          }
        }
      }
    ]
  }
}
